infra/src/create-db.ts

```diff
@@ -0,0 +1,140 @@
+import yargs from 'yargs'
+import { Pool } from 'pg'
+import type { PoolConfig } from 'pg'
+import { SSM } from 'aws-sdk'
+
+import {
+  createExtensions,
+  createDb,
+  createUser,
+  grantPrivileges,
+} from './lib/db'
+import { createOrUpdateSSMParameter, getSSMParameter } from './lib/ssm'
+
+const createPool = (config: PoolConfig) => new Pool(config)
+const getCommandLineArgs = () => {
+  const argv = yargs(process.argv.slice(2))
+    .option('name', {
+      type: 'string',
+      demandOption: true,
+      describe: 'The name of the database to create',
+    })
+    .option('feature', {
+      type: 'string',
+      demandOption: true,
+      describe: 'The name of the feature deployment',
+    })
+    .option('hostname', {
+      type: 'string',
+      demandOption: true,
+      describe: 'RDS hostname',
+    })
+    .option('read-only', {
+      type: 'boolean',
+      describe: 'Create a read-only user',
+      default: false,
+    })
+    .option('extensions', {
+      type: 'array',
+      describe: 'List of PostgreSQL extensions to create',
+      default: [],
+    })
+    .option('ssm-prefix', {
+      type: 'string',
+      describe: 'SSM parameter store prefix',
+      default: '/k8s',
+    })
+    .option('ssm-masterpasword', {
+      type: 'string',
+      describe: 'SSM parameter key for RDS master password',
+      default: '/k8s/vidspyrna/masterpassword',
+    })
+    .option('master-user', {
+      type: 'string',
+      describe: 'RDS root username',
+      default: 'root',
+    })
+    .option('master-database', {
+      type: 'string',
+      describe: 'RDS default database',
+      default: 'postgres',
+    })
+
+    .help()
+    .alias('help', 'h').argv
+
+  return {
+    dbName: argv['name'],
+    featureName: argv['feature'],
+    readOnly: argv['read-only'],
+    dbExtensions: argv.extensions as string[],
+    ssmPrefix: argv['ssm-prefix'],
+    ssmMasterPassword: argv['ssm-masterpasword'],
+    dbMasterUser: argv['master-user'],
+    dbMasterDatabase: argv['master-database'],
+  }
+}
+
+const validateName = (name: string) => /^[a-zA-Z0-9_]+$/.test(name)
+
+const generatePassword = () => Math.random().toString(36).slice(-12)
+
+const main = async () => {
+  const {
+    dbName,
+    readOnly,
+    dbExtensions,
+    dbMasterDatabase,
+    dbMasterUser,
+    featureName,
+    ssmMasterPassword,
+    ssmPrefix,
+  } = getCommandLineArgs()
+
+  if (!validateName(dbName)) {
+    console.error(`Invalid database name: ${dbName}`)
+    return
+  }
+
+  const ssm = new SSM()
+  const masterPassword = await getSSMParameter({
+    ssm,
+    name: ssmMasterPassword,
+  })
+
+  const pool = createPool({
+    user: dbMasterUser,
+    host: process.env.PGHOST,
+    database: dbMasterDatabase,
+    password: masterPassword,
+  })
+
+  try {
+    const dbUser = readOnly ? `${dbName}-ro` : `${dbName}-rw`
```

Comment:
I'm not sure this makes sense to create a new db with only readonly user, as the dev always requires at least write user to run migrations and one would assume an app would at least need write user to write data.  

But it can be useful to get both, as in the sessions service use case where the worker did all the writing, so we could use a read user for the API to add additional security as the worker is an internal process while the API was restful use by our GQL.